### PR TITLE
Serialise indirect calls.

### PR DESCRIPTION
**Pull request by ptersilie**

### CODE REVIEW

**Comment by vext01** [comment]
This LGTM.

**Comment by ptersilie** [comment]
Go for it!

File: llvm/lib/YkIR/YkIRWriter.cpp [hunk 1]
```diff
@@ -479,6 +480,29 @@ class YkIRWriter {
...
+                                 unsigned BBIdx, unsigned &InstIdx) {
+
+    serialiseOpcode(OpCodeIndirectCall);
+    // function type:
+    OutStreamer.emitSizeT(typeIndex(I->getFunctionType()));
+    // callee (operand):
+    serialiseOperand(I, FLCtxt, I->getCalledOperand());
+    // num_args:
+    // (this includes static and varargs arguments)
+    OutStreamer.emitInt32(I->arg_size());
```
Comment (vext01) [inline llvm/lib/YkIR/YkIRWriter.cpp:493]: This doesn't need to happen in this PR, but we should find a way to check if the various `num_` fields we serialise actually fit the data that we populate them with.

e.g. here, does `arg_size()` always fit in 32-bits? etc.

Comment (vext01) [reply llvm/lib/YkIR/YkIRWriter.cpp:493]: (this has been on my mind a few times recently)

**Comment by ltratt** [comment]
I think we can merge this one right now @vext01 ?

**Pull request merged**